bots/notify/src/main/java/org/openjdk/skara/bots/notify/RepositoryWorkItem.java

```diff
@@ -361,16 +365,31 @@ public Collection<WorkItem> run(Path scratchPath) {
             }
 
             boolean hasBranchHistory = !history.isEmpty();
+            int existingCommits = localRepo.getExistingCommits();
             for (var ref : knownRefs) {
                 if (!hasBranchHistory) {
                     log.warning("No previous history found for any branch - resetting mark for '" + ref.name());
-                    for (var listener : listeners) {
-                        log.info("Resetting mark for branch '" + ref.name() + "' for listener '" + listener.name() + "'");
-                        history.setBranchHash(new Branch(ref.name()), listener.name(), ref.hash());
+                    if (existingCommits <= THRESHOLD) {
+                        // In this case, the repo will be considered as a new repo, notify bot will notify start from the first commit
+                        log.info("This is a new repo");
+                        for (var listener : listeners) {
+                            log.info("Resetting mark for branch '" + ref.name() + "' for listener '" + listener.name() + "'");
+                            // Initial the hash for the branches in the first commit, so that the branches will not be treated as 'new'
+                            // and the first commit will be treated as 'update', so we will get notifications
+                            history.setBranchHash(new Branch(ref.name()), listener.name(), new Hash(INITIAL_GIT_HASH));
+                        }
+                    } else {
+                        // In this case, the repo will be considered as an existing repo with history, notify bot will only notify on new commits
+                        log.info("This is an existing repo with history");
```

Comment:
```suggestion
                        log.info("This is an existing repo with history, starting notifications from commits after " + ref.hash());
```
This that log message, I think the comment above becomes redundant.